src/node/wrapper.ts

```diff
@@ -256,18 +266,52 @@ export class WrapperProcess {
       stdio: ["ipc"],
     })
   }
+
+  /**
+   * Wait for a handshake from the child then reply.
+   */
+  private handshake(child: cp.ChildProcess): Promise<void> {
+    return new Promise((resolve, reject) => {
+      const onMessage = (message: Message): void => {
+        logger.debug(`received message from ${child.pid}`, field("message", message))
+        if (message.type === "handshake") {
+          child.removeListener("message", onMessage)
+          child.on("message", (msg) => this._onChildMessage.emit(msg))
+          child.send({ type: "handshake" })
+          resolve()
+        }
+      }
+      child.on("message", onMessage)
+      child.once("error", reject)
+      child.once("exit", (code) => {
+        reject(new ProcessError(`Unexpected exit with code ${code}`, code !== null ? code : undefined))
+      })
+    })
+  }
+}
+
+/**
+ * Process wrapper.
+ */
+export const wrapper =
+  typeof process.env.CODE_SERVER_PARENT_PID !== "undefined"
+    ? new ChildProcess(parseInt(process.env.CODE_SERVER_PARENT_PID))
+    : new ParentProcess(require("../../package.json").version)
+
+export function isChild(proc: ChildProcess | ParentProcess): proc is ChildProcess {
+  return proc instanceof ChildProcess
```

Comment:
Would it be better to just directly do `proc instanceof ChildProcess`?

Reply:
I'm down for that, the main reason I didn't was so I didn't have to export `ChildProcess` since it won't be used anywhere else.

Reply:
All good.